recipes-app/mraa/files/0003-iot2050-add-debugfs-pinmux-and-gpio-chardev-support.patch

```diff
@@ -0,0 +1,701 @@
+From 2562134d8e5c3b62dad50bc760a9cc8e5def9eb8 Mon Sep 17 00:00:00 2001
+From: Benedikt Niedermayr <benedikt.niedermayr@siemens.com>
+Date: Thu, 23 Mar 2023 10:21:58 +0100
+Subject: [PATCH 2/2] iot2050: add debugfs pinmux and gpio chardev support
+
+This patch adds support for multiplexing pins via debugfs rather
+than access memory mapped pad-configuration registers.
+The debugfs pinmux offers the possbility to run mraa on iot2050
+platforms as regular user instead of root by adjusting privileges
+on debugfs files.
+
+Bias settings are currently also configured by accessing pinmux.
+Unfortunatelly a proper upstream-like pinconf usage is currently
+not possible.
```

Comment:
Am I right the you leave the /dev/mem bases muxing support in place? I don't see related code being removed. Is that as fallback on systems not yet support the new scheme?

Reply:
Yes, that was one idea behind that.
The other idea was to remove the /dev/mem access in a separate commit. Depends on discussions and opinions upstream with MRAA folks... 

The fallback mechanism would be a valid/interesting use case if someone uses MRAA as standalone on custom kernels (not directly provided by meta-iot2050).

But in case someone uses the image produced by meta-iot2050 this fallback implementation makes no sense, since all required DTB changes are already included.

Reply:
But I'll go with that fallback mechanism for now